File: src/main/java/ru/stnk/RestTestAPI/entity/Roles.java
package ru.stnk.RestTestAPI.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "roles")
public class Roles implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "Id", nullable = false)
    private Long id;

    @Column(name = "name", unique = true, nullable = false)
    private String name;

    @ManyToMany(fetch = FetchType.LAZY, mappedBy = "roles")
    @JsonIgnore
    private List<User> users;

    public Roles(){}

    public Roles (String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public List<User> getUsers() {
        return users;
    }

    public void setUser(List<User> user) {
        this.users = user;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void addUser(User user) {
        users.add(user);
    }
}

File: src/main/kotlin/ru/stnk/resttestapi/entity/Roles.kt
package ru.stnk.resttestapi.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.security.core.GrantedAuthority
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "roles")
class Roles : GrantedAuthority, Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "Id")
    var id: Long? = null

    @Column(name = "name")
    var name: String = ""

    @ManyToMany(fetch = FetchType.LAZY, mappedBy = "roles")
    @JsonIgnore
    private var users: MutableList<User> = ArrayList()

    constructor() {}

    constructor(name: String) {
        this.name = name
    }

    /*fun getUsers(): List<User>? {
        return users
    }

    fun setUser(user: MutableList<User>) {
        this.users = user
    }

    fun addUser(user: User) {
        users!!.add(user)
    }*/

    override fun getAuthority(): String? {
        return name
    }
}